apiserver/facades/client/modelmanager/register.go

```diff
@@ -66,8 +68,17 @@ func newFacadeV9(ctx facade.Context) (*ModelManagerAPI, error) {
 
 	apiUser, _ := auth.GetAuthTag().(names.UserTag)
 	backend := common.NewUserAwareModelManagerBackend(model, pool, apiUser)
+
+	controllerDB, err := ctx.ControllerDB()
```

Comment:
This violates the pattern we've established, of passing factories to objects depending on databases. We are not materialising these databases during register. They should only be acquired due to an API method _call_.

Reply:
waiting for https://github.com/juju/juju/pull/15945 so we can remove this

Reply:
done, fixed